src/utils/hooks/useKubevirtDataPod/useKubevirtDataPod.ts

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useMemo, useState } from 'react';
+import useWebSocket from 'react-use-websocket';
+
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  consoleFetch,
+  K8sResourceCommon,
+  WatchK8sResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+import useDeepCompareMemoize from '../useDeepCompareMemoize/useDeepCompareMemoize';
+
+import { constructURL, getResourceVersion, registerResourceVersion } from './utils/utils';
+import useKubevirtDataPodFilters from './useKubevirtDataPodFilters';
+
+type UseKubevirtDataPod = <T extends K8sResourceCommon>(
+  watchOptions: WatchK8sResource,
+  filterOptions?: { [key: string]: string },
+) => [T, boolean, Error];
+
+const useKubevirtDataPod: UseKubevirtDataPod = <T extends K8sResourceCommon>(
+  watchOptions: WatchK8sResource,
+  filterOptions?: { [key: string]: string },
+) => {
+  const [data, setData] = useState<T>((<unknown>[]) as T);
+  const [loaded, setLoaded] = useState<boolean>(false);
+  const [error, setError] = useState<Error>(null);
+  const [resourceVersion, setResourceVersion] = useState<string>(null);
+  const query = useKubevirtDataPodFilters(filterOptions);
+  const watchOptionsMemoized = useDeepCompareMemoize<WatchK8sResource>(watchOptions, true);
+  const url = useMemo(
+    () => constructURL(watchOptionsMemoized, query),
+    [query, watchOptionsMemoized],
+  );
+  const [shouldConnect, setShouldConnect] = useState<boolean>(false);
+  const socket = useWebSocket<{ object: K8sResourceCommon; type: string }>(
+    `ws://${window.location.host}${url}`,
+    {
+      onClose: () => console.log('websocket closed kubevirt: ', url),
+      onOpen: () => console.log('websocket open kubevirt: ', url),
```

Comment:
just making sure, this looks intentional, right?

Reply:
yes, same as console, they are printing open and close minus the kubevirt word :)